website/blog/2024/10-14-linkedin-job-scraper-python/index.md

```diff
@@ -0,0 +1,285 @@
+
+    #add all the links to the job listing route
+    await context.add_requests(
+            [Request.from_url(rec, label='job_listing') for rec in hrefs]
+        )
+```
+
+Now that we have the job listings, the next step is to scrape their details.
+
+We'll extract each job’s title, company's name, time of posting, and the link to the job post. Open your dev tools to extract each element using its CSS selector. 
+
+![Inspecting elements](./img/inspect.webp)
+
+After scraping each of the listings, we'll remove special characters from the text to make it clean and push the data to local storage using the `context.push_data` function.
+
+```py
+@router.handler('job_listing')
+async def listing_handler(context: PlaywrightCrawlingContext) -> None:
+    """Handler for job listings."""
+
+    await context.page.wait_for_load_state('load')
+
+    job_title = await context.page.locator('div.top-card-layout__entity-info h1.top-card-layout__title').text_content()
+   
+    company_name  = await context.page.locator('span.topcard__flavor a').text_content()   
+
+    time_of_posting= await context.page.locator('div.topcard__flavor-row span.posted-time-ago__text').text_content()
+
+
+    await context.push_data(
+        {
+            # we are making use of regex to remove special characters for the extracted texts
+
+            'title': re.sub(r'[\s\n]+', '', job_title),
+            'Company name': re.sub(r'[\s\n]+', '', company_name),
+            'Time of posting': re.sub(r'[\s\n]+', '', time_of_posting),
+            'url': context.request.loaded_url,
+        }
+    )
+```
+
+## 3. Creating your application
+
+For this project, we will be using Streamlit for the web application. Before we proceed, we are going to create a new file named `app.py` in your project directory. In addition, ensure you have  [Streamlit](https://docs.streamlit.io/get-started/installation)  installed in your global Python environment before proceeding with this section.
+
+```py
+import streamlit as st
+import subprocess
+
+# Streamlit form for inputs 
+st.title("LinkedIn Job Scraper")
+
+with st.form("scraper_form"):
+    title = st.text_input("Job Title", value="backend developer")
+    location = st.text_input("Job Location", value="newyork")
+    data_name = st.text_input("Output File Name", value="backend_jobs")
+
+    submit_button = st.form_submit_button("Run Scraper")
+
+if submit_button:
+
+    # Run the scraping script with the form inputs
+    command = f"""poetry run python -m linkedin-scraper --title "{title}"  --location "{location}" --data_name "{data_name}" """
+
+    with st.spinner("Crawling in progress..."):
+         # Execute the command and display the results
+        result = subprocess.run(command, shell=True, capture_output=True, text=True)
+
+        st.write("Script Output:")
+        st.text(result.stdout)
+
+        if result.returncode == 0:
+            st.success(f"Data successfully saved in {data_name}.csv")
```

Comment:
Isn't there some way to display the contents of the file in streamlit? That would be a super cool thing to show.

Reply:
will talk with writer and update it later